cmeutils/visualize.py

```diff
@@ -0,0 +1,376 @@
+
+        """
+        return self._unwrap_positions
+
+    @unwrap_positions.setter
+    def unwrap_positions(self, value):
+        if not isinstance(value, bool):
+            raise ValueError(
+                "Set to True or False where "
+                "True uses unwrapped particle positions"
+            )
+        self._unwrap_positions = value
+
+    @property
+    def solid(self):
+        """Set to 1 to use solid colors regardless of light and angle"""
+        return self._solid
+
+    @solid.setter
+    def solid(self, value):
+        self._solid = value
+
+    @property
+    def roughness(self):
+        """Sets the material roughness (ranges from 0.1 to 1)"""
+        return self._roughness
+
+    @roughness.setter
+    def roughness(self, value):
+        self._roughness = value
+
+    @property
+    def specular(self):
+        """Determines the strength of the specular highlights
+        (ranges from 0 to 1)
+        """
+        return self._specular
+
+    @specular.setter
+    def specular(self, value):
+        self._specular = value
+
+    @property
+    def specular_trans(self):
+        """Determines the magnitude of specular light transmission
+        (ranges from 0 to 1)
+        """
+        return self._specular_trans
+
+    @specular_trans.setter
+    def specular_trans(self, value):
+        self._specular_trans = value
+
+    @property
+    def metal(self):
+        """Set to 0 for dielectric material or 1 for metal
+        (ranges from 0 to 1)
+        """
+        return self._metal
+
+    @metal.setter
+    def metal(self, value):
+        self._metal = value
+
+    @property
+    def view_axis(self):
+        """Sets the direction and position of the camera"""
+        return self._view_axis
+
+    @view_axis.setter
+    def view_axis(self, value):
+        # TODO Assert is 1,3  array
+        new_view_axis = np.asarray(value)
+        if new_view_axis.shape != (3,):
+            raise ValueError("View axis must be a 3x1 array")
+        self._view_axis = np.asarray(new_view_axis)
+
+    @property
+    def up(self):
+        """Determines which direction is up"""
+        return self._up
+
+    @up.setter
+    def up(self, value):
+        self._up = value
+
+    @property
+    def height(self):
+        """Acts like a zoom. Larger values zoom out, smaller vaues zoom in"""
+        return self._height
+
+    @height.setter
+    def height(self, value):
+        self._height = value
+
+    @property
+    def camera_position(self):
+        """The camera position.
+        Determined from box dimensions are FresnelGSD.view_axis"""
+        return (self.snapshot.configuration.box[:3] * self.view_axis) - 0.5
+
+    @property
+    def look_at(self):
+        """The direction the camera is facing.
+        By default, uses position directly opposite of camera position
+        """
+        return self.snapshot.configuration.box[:3] * -self.view_axis
+
+    @property
+    def positions(self):
```

Comment:
Looks like some lines in this method are untested.

Reply:
Which ones?